test/open-uri/test_open-uri.rb

```diff
@@ -115,6 +116,13 @@ def test_open_too_many_arg
     assert_raise(ArgumentError) { open("http://192.0.2.1/tma", "r", 0666, :extra) {} }
   end
 
+  def test_tempfile_rest_parameter
+    temp_file = Tempfile.new
+    assert_nothing_raised { open(temp_file, 'a') }
```

Comment:
This still has to be fixed as you can see in travis log:

    ..Leaked file descriptor: TestOpenURI#test_tempfile_rest_parameter: 10 : #<File:/tmp/20170824-55076-tbtyfz>

(If open-uri is not require-ed, or with this patch applied) `open(a_tmpfile)` ends up doing `File.open(a_tmpfile)` (which is equivalent to `File.open(a_tmpfile.to_path)`), and this allocates a new file descriptor in addition to of the File object the Tempfile internally holds.